api/include/opentelemetry/context/runtime_context.h

```diff
@@ -223,15 +223,19 @@ class ThreadLocalContextStorage : public RuntimeContextStorage
 
     Stack() noexcept : size_(0), capacity_(0), base_(nullptr){};
 
-    // Pops the top Context off the stack and returns it.
-    Context Pop() noexcept
+    // Pops the top Context off the stack.
+    void Pop() noexcept
     {
       if (size_ == 0)
       {
-        return Context();
+        return;
       }
+      // Store empty Context before decrementing `size`, to ensure
+      // the shared_ptr object (if stored in prev context object ) are released.
+      // The stack is not resized, and the unused memory would be reutilised
+      // for subsequent context storage.
+      base_[size_ - 1] = Context();
```

Comment:
How could someone get a `shared_ptr` of the `Context` stored in `ThreadLocalContextStorage`?

Reply:
Line 237 is actually resetting the `base[size - 1]` with empty Context() object. This is to ensure that previously-stored shared_ptr<Span> gets released. The new empty Context() is not supposed to be used again.